crates/chain-state/src/memory_overlay.rs

```diff
@@ -87,10 +87,11 @@ impl<N: NodePrimitives> BlockHashReader for MemoryOverlayStateProviderRef<'_, N>
         let mut in_memory_hashes = Vec::new();
         for block in &self.in_memory {
             if range.contains(&block.recovered_block().number()) {
-                in_memory_hashes.insert(0, block.recovered_block().hash());
+                in_memory_hashes.push(block.recovered_block().hash());
                 earliest_block_number = Some(block.recovered_block().number());
             }
         }
+        in_memory_hashes.reverse();
```

Comment:
can we add a note here, explaining that we need to reverse the order here so that this is in desc order